netbox/extras/scripts.py

```diff
@@ -331,6 +354,21 @@ def job_timeout(self):
     def scheduling_enabled(self):
         return getattr(self.Meta, 'scheduling_enabled', True)
 
+    @property
+    def filename(self):
+        return inspect.getfile(self.__class__)
+
+    @property
+    def source(self):
+        return inspect.getsource(self.__class__)
+
+    @property
+    def is_valid(self):
+        """
+        Indicates whether the report can be run.
+        """
+        return bool(self.test_methods)
```

Comment:
I'm not sure this makes sense to keep around. IIRC `is_valid` was used to determine whether a Report had at least one test method defined. This is not a prerequisite for running a script.